### PR TITLE
Make public span-based key in CreateHMAC.

### LINKED ISSUE #35869: API Proposal: Make IncrementalHash.CreateHMAC ReadOnlySpan<byte> key public
Since HKDF was added, IncrementalHash has had support for `ReadOnlySpan<byte>` HMAC keys, but it isn't made public.

The functionality is already there, it just needs to be changed from internal to public and given proper test coverage.

## Proposal:

```csharp
 namespace System.Security.Cryptography {
     public sealed class IncrementalHash : IDisposable {
        // existing method changed from internal to public
        public static IncrementalHash CreateHMAC(HashAlgorithmName hashAlgorithm, ReadOnlySpan<byte> key);
     }
 }
```

/cc @JimBobSquarePants

### PR DESCRIPTION
**Pull request by vcsjones**
Fixes #35869

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/IncrementalHash.cs [hunk 1]
```diff
@@ -354,7 +354,29 @@ public static IncrementalHash CreateHMAC(HashAlgorithmName hashAlgorithm, byte[]
             return CreateHMAC(hashAlgorithm, (ReadOnlySpan<byte>)key);
         }
 
-        internal static IncrementalHash CreateHMAC(HashAlgorithmName hashAlgorithm, ReadOnlySpan<byte> key)
+        /// <summary>
```
Comment (vcsjones) [inline src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/IncrementalHash.cs:357]: Docs are a lift from the `byte[]` overload.

**Review by bartonjs — APPROVED** [review]

**Comment by vcsjones** [comment]
Oops. Mis-tap of the thumb.

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
Thanks

**Comment by bartonjs** [comment]
The only failures are in RuntimeInformation on wasm, doesn't make sense that this would be related.

```
System.Runtime.InteropServices.RuntimeInformationTests.PlatformVersionTests.IsOSPlatformOrLater_ReturnsTrue_ForCurrentOS
System.ArgumentOutOfRangeException : Illegal enum value: Other. (Parameter 'platform')
  Actual value was Other.
 at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

**Pull request merged**